tests/helpers/zos_lineinfile_helper.py

```diff
@@ -159,3 +186,126 @@ def DsGeneralResultKeyMatchesRegex(test_name, ansible_zos_module, test_env, test
+        # call line infile to see results      
+        results = hosts.all.zos_lineinfile(**test_info)
+        pprint(vars(results))
+        if test_env["ENCODING"] == 'IBM-1047':
+            cmdStr =r"""cat "//'{0}'" """.format(test_info["path"])
+            results = hosts.all.shell(cmd=cmdStr)
+            pprint(vars(results))
+            for result in results.contacted.values():
+                print()
+                assert result.get("stdout").replace('\n', '').replace(' ', '') == expected.replace('\n', '').replace(' ', '')
+    finally:
+        # extract pid
+        ps_list_res = hosts.all.shell(cmd="ps -e | grep -i 'pdse-lock'")
+
+        # kill process - release lock - this also seems to end the job
+        pid = list(ps_list_res.contacted.values())[0].get('stdout').strip().split(' ')[0]
+        hosts.all.shell(cmd="kill 9 {0}".format(pid.strip()))
+        # clean up c code/object/executable files, jcl
+        hosts.all.shell(cmd='rm -r /tmp/disp_shr')
+        # remove pdse
+        hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="absent")
+
+
+def DsGeneralForceFail(ansible_zos_module, test_env, test_info):
+    MEMBER_1, MEMBER_2 = "MEM1", "MEM2"
+    hosts = ansible_zos_module
+    test_info["path"] = DEFAULT_DATA_SET_NAME+"({0})".format(MEMBER_2)
+    if test_env["ENCODING"]:
+        test_info["encoding"] = test_env["ENCODING"]
+    try:
+        # set up:
+        # create pdse
+        hosts.all.zos_data_set(name=DEFAULT_DATA_SET_NAME, state="present", type="pdse", replace=True)
+        # add members
+        hosts.all.zos_data_set(
+            batch=[
+                {
+                    "name": DEFAULT_DATA_SET_NAME + "({0})".format(MEMBER_1),
+                    "type": "member",
+                    "state": "present",
+                    "replace": True,
+                },
+                {
+                    "name": DEFAULT_DATA_SET_NAME + "({0})".format(MEMBER_2),
+                    "type": "member",
+                    "state": "present",
+                    "replace": True,
+                },
+            ]
+        )
+        # copy/compile c program and copy jcl to hold data set lock for n seconds in background(&)
+        hosts.all.zos_copy(content=c_pgm, dest='/tmp/disp_shr/pdse-lock.c', force=True)
+        hosts.all.zos_copy(
+            content=call_c_jcl.format(DEFAULT_DATA_SET_NAME, MEMBER_1),
+            dest='/tmp/disp_shr/call_c_pgm.jcl',
+            force=True
+        )
+        hosts.all.shell(cmd="xlc -o pdse-lock pdse-lock.c", chdir="/tmp/disp_shr/")
+        # submit jcl
+        hosts.all.shell(cmd="submit call_c_pgm.jcl", chdir="/tmp/disp_shr/")
+        # pause to ensure c code acquires lock
+        time.sleep(5)  
+        # call line infile to see results      
+        results = hosts.all.zos_lineinfile(**test_info)
+        pprint(vars(results))
+        for result in results.contacted.values():
+            assert result.get("changed") == 0
```

Comment:
see previous comment about comparing `bool` to `bool` instead of `bool` to `int`